### PR TITLE
fix(TDP-4889): fix quality bar on math action --> add TI

### PR DESCRIPTION
**Pull request by tlnd-jbrun**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4889

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/PreparationStep.java [hunk 1]
```diff
@@ -266,8 +267,12 @@ public void thePreparationShouldHaveThefollowingInvalidCells(String preparationN
         String invalidCells = parameters.get("invalidCells");
 
         HashMap values = (LinkedHashMap<String, String>) datasetContent.records.get(Integer.parseInt(columnNumber));
```
Comment (hmdebenque) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/PreparationStep.java]: but why a cast to LinkedHashMap to use it as HashMap ?

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/PreparationStep.java [hunk 2]
```diff
@@ -266,8 +267,12 @@ public void thePreparationShouldHaveThefollowingInvalidCells(String preparationN
...
-
-        assertEquals(invalidCells, values.get("__tdpInvalid"));
+        if (!invalidCells.equals(StringUtils.EMPTY)) {
+            assertEquals(invalidCells, values.get("__tdpInvalid"));
```
Comment (hmdebenque) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/PreparationStep.java]: please use a constant for `__tdpInvalid`

File: dataprep-test-api/src/test/resources/features/DeleteInvalidActionTest.feature [hunk 3]
```diff
@@ -11,6 +11,10 @@ Feature: Check some features of the Duplicate Action and the Delete Invalid Acti
     Given I create a preparation with name "phoneNumber_delete_invalid_prep", based on "phoneNumber_delete_invalid_dataset" dataset
     Then The characteristics of the preparation "phoneNumber_delete_invalid_prep" match:
       | sample_records_count | 6 |
+    Then The preparation "phoneNumber_delete_invalid_prep" should have the following quality bar characteristics on the column number "2":
```
Comment (hmdebenque) [inline dataprep-test-api/src/test/resources/features/DeleteInvalidActionTest.feature]: why testing the same bug fix in several tests ?

File: dataprep-test-api/src/test/resources/features/NegateBooleanActionTest.feature [hunk 4]
```diff
@@ -10,6 +10,10 @@ Feature: Check some features of Negate Boolean Action
     Given I upload the dataset "/data/NegateBooleanAction.txt" with name "NegateBooleanAction_dataset"
     Then I wait for the dataset "NegateBooleanAction_dataset" metadata to be computed
     Given I create a preparation with name "NegateBooleanAction_prep", based on "NegateBooleanAction_dataset" dataset
+    Then The preparation "NegateBooleanAction_prep" should have the following quality bar characteristics on the column number "0":
```
Comment (hmdebenque) [inline dataprep-test-api/src/test/resources/features/NegateBooleanActionTest.feature]: same here: why duplicate testing ?

File: dataprep-test-api/src/test/resources/features/DeleteInvalidActionTest.feature (hunk 3 above)
Comment (tlnd-jbrun) [reply dataprep-test-api/src/test/resources/features/DeleteInvalidActionTest.feature]: it is not the same action, not the same bug TDP-6272. Duplicate is not a Math action

File: dataprep-test-api/src/test/resources/features/NegateBooleanActionTest.feature (hunk 4 above)
Comment (tlnd-jbrun) [reply dataprep-test-api/src/test/resources/features/NegateBooleanActionTest.feature]: Not the same thing as i Said 2 different actions, not the same behaviours, so it does not test the same thing

**Pull request merged**